repose-aggregator/components/filters/destination-router-filter/src/main/scala/org/openrepose/filters/destinationrouter/DestinationRouterFilter.scala

```diff
@@ -0,0 +1,116 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.filters.destinationrouter
+
+import java.util
+import java.util.concurrent.TimeUnit
+import javax.inject.{Inject, Named}
+import javax.servlet._
+import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
+
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import org.openrepose.commons.config.manager.UpdateListener
+import org.openrepose.commons.utils.http.CommonRequestAttributes
+import org.openrepose.commons.utils.servlet.http.RouteDestination
+import org.openrepose.core.filter.FilterConfigHelper
+import org.openrepose.core.filters.DestinationRouter
+import org.openrepose.core.services.config.ConfigurationService
+import org.openrepose.core.services.reporting.metrics.{MeterByCategorySum, MetricsService}
+import org.openrepose.filters.routing.servlet.config.DestinationRouterConfiguration
+
+@Named
+class DestinationRouterFilter @Inject()(configurationService: ConfigurationService, metricsService: MetricsService)
+  extends Filter with UpdateListener[DestinationRouterConfiguration] with LazyLogging {
+
+  private final val DefaultConfigFileName = "destination-router.cfg.xml"
+  private final val SchemaFilePath = "/META-INF/schema/config/destination-router-configuration.xsd"
+
+  private val mbcsRoutedReponse: Option[MeterByCategorySum] = Option(metricsService) flatMap { ms =>
```

Comment:
I thought we decided we didn't have to worry about the metrics service being null these days.